upgrade-2.x.sh

```diff
@@ -0,0 +1,204 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+
+target_version=$1
+preferred_hostos_version=2.0.7
+
+. /etc/os-release
+
+# Log function helper
+function log {
+    # Address log levels
+    case $1 in
+        ERROR)
+            loglevel=ERROR
+            shift
+            ;;
+        WARN)
+            loglevel=WARNING
+            shift
+            ;;
+        *)
+            loglevel=LOG
+            ;;
+    esac
+    endtime=$(date +%s)
+    if [ "z$LOG" == "zyes" ] && [ -n "$LOGFILE" ]; then
+        printf "[%09d%s%s\n" "$((endtime - starttime))" "][$loglevel]" "$1" | tee -a "$LOGFILE"
+    else
+        printf "[%09d%s%s\n" "$((endtime - starttime))" "][$loglevel]" "$1"
+    fi
+    if [ "$loglevel" == "ERROR" ]; then
+        /usr/bin/resin-device-progress --percentage 100 --state "ResinOS: Update failed."
+        exit 1
+    fi
+}
+
+# Test if a version is greater than another
+function version_gt() {
+    test "$(echo "$@" | tr " " "\n" | sort -V | head -n 1)" != "$1"
+}
+
+# Log timer
+starttime=$(date +%s)
+
+# Check board support
+case $SLUG in
+    beaglebone*)
+        binary_type=arm
+        ;;
+    raspberry*)
+        binary_type=arm
+        ;;
+    *)
+        log ERROR "Unsupported board type $SLUG."
+esac
+
+if [ -n "$target_version" ]; then
+    target_version=$(echo "$target_version" | tr + _)
+    case $target_version in
+        2.*)
+            log "Target OS version \"$target_version\" OK."
+            ;;
+        *)
+            log ERROR "Target OS version \"$target_version\" not supported."
+            ;;
+    esac
+else
+    log ERROR "No target OS version specified."
+fi
+
+# Check host OS version
+case $VERSION in
```

Comment:
I was little confused about who sets this variable. Maybe a comment that it comes from os-release would help. Or actually when sourcing we could say what variables we are relying on from that file. Just to be clear.